drivers/temperature/adt7420/adt7420.c

```diff
@@ -419,17 +411,36 @@ int adt7420_spi_reg_read(struct adt7420_dev *dev,
  * @return register_value  - Value of the register.
 *******************************************************************************/
 int adt7420_i2c_reg_read(struct adt7420_dev *dev,
-			 uint8_t register_address, uint8_t *data)
+			 uint8_t register_address, uint16_t *data)
 {
-	uint8_t register_value = 0;
+	uint8_t data_buffer[3] = { 0, 0 };
+	uint8_t num_bytes;
+
+	switch (register_address) {
+	case ADT7420_REG_STATUS:
```

Comment:
can't we use here `no_os_field_get(ADT7320_L16, register_address)` instead of switch...case?

Reply:
Yes, the approach similar to SPI registers can be done. The same change has been done now